documentation/codeQuality/README.md

```diff
@@ -1,2 +1,233 @@
+}
+```
+8. Now copy and run the second command
+    * For Windows user, run the second command on GitBash (there was a problem when I tried to run it with cmd and powerShell).
+
+9. After the second command finished executing, reload the SonarCloud's website. It will display analytic results to you.
+
+![analytic results](https://user-images.githubusercontent.com/11821799/46253852-e1b73000-c4af-11e8-9309-8a9e7676958e.png)
+
+10. What just happened ? We have just push our code in the working branch to SonarCloud to analyze the standard of our code quality.
+
+11. Is the code that is being analyze comes from the origin or local changes ? Try adding the following lines into your App.java. Save the file without committing or pushing the code change.
+```java
+    public static void main(String[] args) {
+        String amount = args[0];
+
+        // Add this line
+        double result = 1.1 + 1.2;
+
+        CurrencyFormatter currencyFormatter = new CurrencyFormatter();
+
+        System.out.println(currencyFormatter.format(amount));
+    }
+```
+12. Rerun the analyze command again !
+
+## More problems !
+We have said early on that SonarCloud does not only analyze your code base, however it actually reports the unit test coverage of our project. 
+
+Having unit test coverage helps us to have visibility on how trustful our software is. With test coverage report, we should be able to easily tell where unit test are missing and priority them in the development team.
+
+However, right now, even we have added some unit test in the beginning of this workshop SonarCloud still reports 0% coverage for our project.
+
+![no coverage](https://user-images.githubusercontent.com/11821799/46254263-98b6aa00-c4b6-11e8-8686-3fd9d019ace6.png)
+
+## Get test coverage to display
+If we take a look at our build.gradle, we will see that we do not have any plugin for code coverage. That is why SonarCloud were not able to pick up the result and display the report to us.
+
+### Let's add JaCoCo (Java Code Coverage)
+JaCoCo is a gradle's plugin that helps provide code coverage metrics to us. You can read more about JaCoCo [here](https://docs.gradle.org/current/userguide/jacoco_plugin.html).
+
+1. Go to build.gradle and then add JaCoCo into plugins section. Here is what our plugin section looks like.
+```
+plugins {
+    // Apply the java plugin to add support for Java
+    id 'java'
+
+    // Apply the application plugin to add support for building an application
+    id 'application'
+
+    id "org.sonarqube" version "2.6"
+
+    // Add this line
+    id "jacoco"
+}
+```
+
+2. To see the test report, run the following command
+```
+gradle test jacocoTestReport
+```
+
+3. By default, you test report will be located under .build/reports/jacoco
+
+4. Now let's return the SonarCloud analyze command. Your coverage result should be present on the SonarCloud's site.
+
+5. In Addition, you can fix the version of JaCoCo by adding this section into build.gradle
```

Comment:
may be change 

fix -> specify

When read this sentence for the first time, I thought I am going to fix something.

Reply:
fixed